---
author: Krtin Juneja (KJUNEJA@falcon.bentley.edu)
---

From `statsmodels`, we use the `solve_power` function in the `TTestIndPower` class.  That function  embodies a relationship among five variables; you provide any four of them and it will compute the fifth to satisfy the relationship.  Let's get started by importing the package and create a `TTestIndPower` object.

In [4]:
from statsmodels.stats.power import TTestIndPower
analysis = TTestIndPower()

For this example, let's say that:

 * You plan to create a balanced $4\times2$ factorial experiment with 32 subjects.
 * You expect the effect size for the main effect of factor A to be medium (0.25 according to Cohen's 1988 text).
 * You want to know the expected power for the test of a main effect of factor A.
 * Your significance level is $\alpha=0.05$.

We proceed as follows.

In [5]:
obs = 32       # number of subjects (or observations)
effect = 0.25  # alpha, the effect size 
alpha = 0.05   # significance level
ratio = 1      # the ratio of the number of observations in one sample to the other

# We leave power unspecified, so that solve_power will compute it for us:
analysis.solve_power( effect, power=None, nobs1=obs, ratio=0.8, alpha=alpha )

0.15266275254949752

The power is 0.15 which means that the probability of rejecting the null hypothesis when in fact it is false OR the probability of avoiding a Type II error is 0.15.